src/java/org/apache/cassandra/db/filter/DataLimits.java

```diff
@@ -934,7 +934,7 @@ protected Row applyToStatic(Row row)
                 // It's possible that we're "done" if the partition we just started bumped the number of groups (in
                 // applyToPartition() above), in which case Transformation will still call this method. In that case, we
                 // want to ignore the static row, it should (and will) be returned with the next page/group if needs be.
-                if (isDone())
+                if (enforceLimits && isDone())
```

Comment:
@adelapena It doesn't look like anything in `TestConsistency::test_group_by_srp` fails if we remove this. (i.e. It looks like the new logic is correct, just doesn't have any coverage.)

Reply:
Right, just added an in-JVM dtests for each condition.